guava/src/com/google/common/escape/CharEscaperBuilder.java

```diff
@@ -108,9 +113,13 @@ public CharEscaperBuilder addEscapes(char[] cs, String r) {
    *
    * @return a "sparse" array that holds the replacement mappings.
    */
+  @SuppressWarnings(value = {"array.length.negative",//int max is constant -1, max + 1 = 0 is non negative
+            "array.access.unsafe.low",//Character types are non negative: https://github.com/kelloggm/checker-framework/issues/192
+            "enhancedfor.type.incompatible",}//map is declared after result array
```

Comment:
It's acceptable to reorder statements if they don't depend on each other and a warning is removed, like in this case.